### ISSUE TITLE
fetch returns empty when the same template calls it more than once 

### ISSUE THREAD

**Issue by lsycxyj** [umakantp/jsmart#11]
I call fetch successfully at the first time under nodejs, but after that it always returns empty string. I think there may be bugs due to caching after I read the source of jSmart.

``` javascript
jSmart.prototype.getTemplate = function(name){
    return xxx; //some parent content
}

var t1 = new jSmart('xxx'); //a template contains "extends"
var first = t1.fetch({xxx:'xxx'});//first is not empty
var t2 = new jSmart('xxx');//Same template as t1
var second = t2.fetch({xxx:'xxx'});//Same content as t1, but second is empty now
```

**Comment by Avcajaraville** [comment]
Im having the same issue here, all the problems comes with the nocache thing... This is totally stoping me from further development. 

Is this repo. even maintained ?

**Comment by hereandnow** [comment]
I took a look into it, it should probably be fairly easy to fix:

just enable the "nocache" option param when [calling getTemplate in the built-in extend parse function](https://github.com/umakantp/jsmart/blob/master/jsmart.js#L800):

before:

    getTemplate(trimQuotes(params.file?params.file:params[0]), tree);

after:

    getTemplate(trimQuotes(params.file?params.file:params[0]), tree, true);

**Comment by salex772** [comment]
Yes, I had the same issue. Thanks @hereandnow

**Comment by hereandnow** [comment]
@salex772: you could use @Avcajaraville's fork https://github.com/Avcajaraville/grunt-jsmart

it is at least a little maintained!

**Comment by Avcajaraville** [comment]
@salex772 The fork @hereandnow is referring is this one --> https://github.com/Avcajaraville/jsmart

The other one was a fork for grunt plugin.

So far, its doing the job for me, and since we are using it for some development process, whenever we find a missing feature, we will keep enhancing it.

**Comment by salex772** [comment]
@Avcajaraville What about UMD and upload these all to npm?

**Comment by Avcajaraville** [comment]
@salex772 It have also UMD support. About uploading this to npm... yeah, will do that on the next days, since every day there is more people using it.

On the meantime, you can add a dependency to your package.json like this:

````
  ...
  "dependencies": {
    "jsmart": "Avcajaraville/jsmart#master"
  },
  ...
````

And then just require it as usually: `var jsmart = require( 'jsmart' );`

Hope this helps you.

**Comment by salex772** [comment]
No, I prefer to wait for npm version and BTW, it should work with Webpack.

**Comment by umakantp** [comment]
This has been fixed in  the release v2.15.1
